examples/drawing/plot_spring_layout.py

```diff
@@ -0,0 +1,80 @@
+"""
+===================
+Spring Layout
+===================
+
+Draw graphs using the three different spring layout algorithms.
+
+The spring layout is typically generated by the Fruchterman--Reingold force-directed algorithm.
+This algorithm treats edges as springs holding nodes close while treating nodes as repelling objects
+and simulates the system until it reaches equilibrium.
+The algorithm also terminates when it reaches a maximum number of iterations.
+
+NetworkX offers mainly three different kinds of methods based on the same theoretical foundation:
+
+* `nx.spring_layout(G, method="force")`
+
+  * The default for graphs with fewer than 500 nodes in `nx.spring_layout`.
+  * Direct implementation of the Fruchterman--Reingold force-directed algorithm.
+  * Can handle negative edge weights.
+
+* `nx.spring_layout(G, method="energy")`
+
+  * The default for graphs with more than or equal to 500 nodes in `nx.spring_layout`.
+  * It solves an energy-based optimization problem, taking the absolute value of negative edge weights.
+  * Generally produces higher-quality layouts compared to `force`.
+  * Uses gravitational forces acting on each connected component to prevent divergence.
+
+* `nx.nx_agraph.graphviz_layout(G, prog="sfdp")`
+
+  * Uses `sfdp` from GraphViz to compute the layout.
+  * Can also be done with `nx.nx_pydot.graphviz_layout(G, prog="sfdp")`.
+  * Employs the multilevel approach to apply force-directed graph drawing progressively.
+  * Requires separate installation of GraphViz. For details, see
+    `here <https://networkx.org/documentation/stable/reference/drawing.html#module-networkx.drawing.nx_agraph>`_.
+  * Especially fast for large graphs.
+"""
+
+import time
+import matplotlib.pyplot as plt
+import matplotlib.patches as mpatches
+import networkx as nx
+
+graphs = [
+    (nx.cycle_graph(100), "cycle"),
+    (nx.grid_2d_graph(25, 25), "grid_2d"),
```

Comment:
Would you be comfortable shrinking the grid2d example in size? As it stands, it makes this one of the longer running gallery examples... I think if we drop it even to 15x15 you still get the behavior you want to illustrate (i.e. energy is better than force) and the runtime is much more manageable!

```suggestion
    (nx.grid_2d_graph(15, 15), "grid_2d"),
```